griptape/tasks/prompt_task.py

```diff
@@ -67,14 +67,14 @@ def prompt_stack(self) -> PromptStack:
             self.generate_system_template(self)
         )
 
-        if memory:
-            memory.add_to_prompt_stack(stack)
-
         stack.add_user_input(self.input.to_text())
 
         if self.output:
             stack.add_assistant_input(self.output.to_text())
 
+        if memory:
+            memory.add_to_prompt_stack(stack, 1)
```

Comment:
May be add a comment explaining that this injects memory after the system input (`stack.inputs[index:index]`), hence the index of `1` here.